app/src/main/java/com/android/sample/model/recipe/FirestoreRecipesRepository.kt

```diff
@@ -0,0 +1,174 @@
+package com.android.sample.model.recipe
+
+import android.util.Log
+import com.android.sample.resources.C.Tag.FIRESTORE_COLLECTION_NAME
+import com.android.sample.resources.C.Tag.FIRESTORE_RECIPE_AREA
+import com.android.sample.resources.C.Tag.FIRESTORE_RECIPE_CATEGORY
+import com.android.sample.resources.C.Tag.FIRESTORE_RECIPE_DIFFICULTY
+import com.android.sample.resources.C.Tag.FIRESTORE_RECIPE_INGREDIENTS
+import com.android.sample.resources.C.Tag.FIRESTORE_RECIPE_INSTRUCTIONS
+import com.android.sample.resources.C.Tag.FIRESTORE_RECIPE_MEASUREMENTS
+import com.android.sample.resources.C.Tag.FIRESTORE_RECIPE_NAME
+import com.android.sample.resources.C.Tag.FIRESTORE_RECIPE_PICTURE_ID
+import com.android.sample.resources.C.Tag.FIRESTORE_RECIPE_PRICE
+import com.android.sample.resources.C.Tag.FIRESTORE_RECIPE_TIME
+import com.android.sample.resources.C.Tag.LIMIT_MUST_BE_POSITIVE_MESSAGE
+import com.android.sample.resources.C.Tag.UNSUPPORTED_MESSAGE
+import com.google.android.gms.tasks.Task
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+
+class FirestoreRecipesRepository(private val db: FirebaseFirestore) : RecipesRepository {
```

Comment:
Nice work on this! The structure is clean, and everything seems to be covered well.

**A Few Quick Suggestions**:
1. **Extra Null Safety in `documentToRecipe`**: Might be helpful to add a few early null checks for required fields so it fails quickly if something’s missing:
   ```kotlin
   val name = document.getString(FIRESTORE_RECIPE_NAME) ?: return null
   ```

2. **Consider Batch Ops**: If we’re updating/deleting multiple recipes, batching them could speed things up.

3. **More Detailed Error Messages**: Including IDs (like `mealID` in `search`) in error messages helps track specific issues faster.

Overall, solid job!

Reply:
Well it is already done I think

Reply:
I used a wrong example sorry. I was talking for others fields where they arent any extra null safety like for the `FIRESTORE_RECIPE_CATEGORY`

Reply:
Ok will do that